lib/erlef_web/controllers/events_controller.ex

```diff
@@ -5,7 +5,10 @@ defmodule ErlefWeb.EventController do
   alias Erlef.Events
 
   def index(conn, _params) do
-    render(conn)
+    render(conn,
+      gcal_api_key: Application.get_env(:erlef, :gcal_api_key),
```

Comment:
do we use releases when deploying?

Reply:
Hi @pdgonzalez872,

I am no aware we are using releases.

Can you please explain your concerns about using the code you pointed out?

Thanks,

Best regards.

Reply:
@joaquinalcerro hi! Sorry, I should have prefaced it with `Nit/Question` since it is not really important/relevant here. The reason I brought it up is that there is a difference between compile time and runtime system variables and sometimes it can cause confusion. It is not the case here, just made me think about `releases` since it is one of the place where the confusion usually comes up.

Reply:
Cool. Thanks for the feedback.

Reply:
That is correct we are using releases. In this case getting the env var from the system is preferred, otherwise we bake the key into the release. So a simple change to `System.get_env/1` would be good.

Reply:
@starbelly @pdgonzalez872 

I just wanted to point out the fact that I used the System.get_env/1 function in the config.exs file in order to have a single place where System calls for environment variables are called:

```
config :erlef,
  gcal_api_key: System.get_env("GCAL_API_KEY"),
  gcal_id: System.get_env("GCAL_ID")
```

Then I used the Application.get_env/2 function in the events_controller.ex as you point out.

If this is okay with you, I can leave it as is or I can just change the code as @starbelly last comment. Also, can you please let me know it there are additional changes required to send final update.

Best regards,

Reply:
Right, this area is always a bit of confusion so let me see if I can demonstrate:

I'm going to add `config :erlef, :test, System.get_env("FOO")` to config.exs then assemble a release:


### Set our env var to "foo"

```
$ export FOO="foo" 
```

### Compile a release

```
$  mix release
===> Compiling ... 
...
* assembling erlef-0.1.0 on MIX_ENV=dev
...
To list all commands:

    _build/dev/rel/erlef/bin/erlef
```

### Start up the release with iex attached and get our app config

```
 _build/dev/rel/erlef/bin/erlef start_iex
Erlang/OTP 22 [erts-10.4.4] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe] [dtrace]

Interactive Elixir (1.9.1) - press Ctrl+C to exit (type h() ENTER for help)
iex(erlef@boris)1> Application.get_env(:erlef, :test)
"foo"
iex(erlef@boris)2>
BREAK: (a)bort (c)ontinue (p)roc info (i)nfo (l)oaded
       (v)ersion (k)ill (D)b-tables (d)istribution
q
```

Ok, so far so good, we are able to get the value we set for FOO in the system environment, but let's change that env var and start up again. 

```
$ export FOO="bar"
$_build/dev/rel/erlef/bin/erlef start_iex
Erlang/OTP 22 [erts-10.4.4] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe] [dtrace]

Interactive Elixir (1.9.1) - press Ctrl+C to exit (type h() ENTER for help)
iex(erlef@Bryans-MacBook-Pro)1> Application.get_env(:erlef, :test)
"foo"
iex(erlef@Bryans-MacBook-Pro)2>
```

 It is still `"foo"` 😢 . This is because the env var was grabbed at compile time (not runtime) and baked in. This is why we want to use `System.get_env/1` in our functions for such configuration. It should be noted that elixir 1.9 introduced the convention of `config/releases.exs` which you can read about more [here](https://hexdocs.pm/mix/1.9.0/Mix.Tasks.Release.html#module-runtime-configuration). 

 I am open to using this new convention or just calling `System.get_env/1` in the controller. Either way it must be changed I'm afraid.

Reply:
Thanks for the clarification and ample explanation, don't worry I am still learning.

I will change the code and submit a new pull request.

Best regards,

Reply:
No worries 😁  I love teaching 👍